src/esp/gfx/RenderCamera.cpp

```diff
@@ -209,11 +210,20 @@ esp::geo::Ray RenderCamera::unproject(const Mn::Vector2i& viewportPosition) {
           Magnum::Vector2{1.0f},
       1.0};
 
+  // compute the far plane
+  auto zFar =
+      Mn::Frustum::fromMatrix(projectionMatrix() * cameraMatrix()).far();
+  float farDistance = zFar[3] / zFar.xyz().length();
+
   ray.direction =
       ((object().absoluteTransformationMatrix() * projectionMatrix().inverted())
```

Comment:
(Ignore if this utility isn't meant to be used in performance-sensitive scenarios.)

The inverted projection matrix could maybe get cached, otherwise it'll be the most time consuming calculation of the whole function.

Reply:
Good point, caching makes sense.